lib/appmap/handler/net_http.rb

```diff
@@ -0,0 +1,107 @@
+# frozen_string_literal: true
+
+require 'appmap/event'
+
+module AppMap
+  module Handler
+    class HTTPClientRequest < AppMap::Event::MethodEvent
+      attr_accessor :request_method, :url, :params, :headers
+
+      def initialize(http, request)
+        super AppMap::Event.next_id_counter, :call, Thread.current.object_id
+
+        path, query = request.path.split('?')
+        query ||= ''
+
+        protocol = http.use_ssl? ? 'https' : 'http'
+        port = if http.use_ssl? && http.port == 443
+          nil
+        elsif !http.use_ssl? && http.port == 80
+          nil
+        else
+          ":#{http.port}"
+        end
+
+        url = [ protocol, '://', http.address, port, request.path ].compact.join
```

Comment:
Let's remove the query if it's going to be passed as params. I've updated the https://github.com/applandinc/appmap/pull/18 proposal.